projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/vdk_jupyterlab_extension/handlers.py

```diff
@@ -111,16 +112,20 @@ def post(self):
             self.finish(json.dumps({"message": f"{e}", "error": "true"}))
 
 
-class ConvertJobToNotebookHandler(APIHandler):
+class ConvertJobHandler(APIHandler):
     """
-    Class responsible for handling POST request for converting a Data Job to Notebook given the Rest API URL
-    and the path to its directory
+    Class responsible for handling POST request for transforming a directory type Data job(with .py and .sql files)
+    to a notebook type data job
     """
 
     @tornado.web.authenticated
     def post(self):
-        # TODO fix this as part of the implementation
-        print("Successfully connected to the Convert Job To Notebook handler!")
+        input_data = self.get_json_body()
+        try:
+            message = json.dumps(VdkUI.convert_job(input_data[VdkOption.PATH.value]))
```

Comment:
It might be a good idea to put these `json.dumps()` calls inside the `convert_job` and `finish` methods. Unless those methods are used for something else, this would make the code much more readable.

Reply:
@DeltaMichael it is used in all the other handlers too so i think making one different is not good but idk

Reply:
Please open an improvement ticket for it, like we discussed.

Reply:
@DeltaMichael https://github.com/vmware/versatile-data-kit/issues/2438